tools/SourceKit/include/SourceKit/Core/LangSupport.h

```diff
@@ -384,22 +384,36 @@ struct ParentInfo {
   StringRef USR;
 };
 
+struct CursorSymbolInfo {
+  UIdent Kind;
+  UIdent DeclarationLang;
+  llvm::SmallString<32> Name;
+  llvm::SmallString<64> USR;
+  StringRef GroupName;
+  /// Non-empty if the symbol was imported from a clang module.
+  llvm::SmallString<32> ModuleName;
+  /// Non-empty if a generated interface editor document has previously been
+  /// opened for the module the symbol came from.
+  StringRef ModuleInterfaceName;
+  /// Set only if the declaration has a source location.
+  StringRef Filename;
+  /// This is an (offset,length) pair. It is set only if the declaration has a
+  /// source location.
+  llvm::Optional<std::pair<unsigned, unsigned>> DeclarationLoc = None;
+  bool IsSystem = false;
+};
+
 struct CursorInfoData {
   // If nonempty, a proper Info could not be resolved (and the rest of the Info
   // will be empty). Clients can potentially use this to show a diagnostic
   // message to the user in lieu of using the empty response.
   StringRef InternalDiagnostic;
 
-  UIdent Kind;
-  UIdent DeclarationLang;
-  StringRef Name;
-  StringRef USR;
+  CursorSymbolInfo Symbol;
   StringRef TypeName;
   StringRef TypeUSR;
   StringRef ContainerTypeUSR;
   StringRef DocComment;
```

Comment:
Why were these fields not part of `Symbol`?  If you want to find documentation of the initializer for example you would want DocComment.  Same for the ones below such as the annotated declaration.  I'm wondering why we don't just pass multiple `CursorInfoData`.

Reply:
That's fair, I'll likely move all the fields into `CursorSymbolInfo` and then have a `Results` or something in `CursorInfoData` along with `AvailableActions` and `InternalDiagnostic`.